File: SmartParking.Domain/Entities/ParkingSession.cs
using SmartParking.Domain.Common;

namespace SmartParking.Domain.Entities;

public class ParkingSession : BaseEntity
{
    public Guid UserId { get; set; }
    public Guid ParkingLotId { get; set; }
    public ParkingLot ParkingLot { get; set; } = null!;
    public Guid? ParkingSpaceId { get; set; }
    public ParkingSpace? ParkingSpace { get; set; }
    public string VehiclePlate { get; set; } = string.Empty;
    public DateTime StartedAt { get; set; }
    public DateTime? EndedAt { get; set; }
    public decimal? TotalAmount { get; set; }
    public string PaymentStatus { get; set; } = "pending";
}

File: lib/models/parking_session.dart
class ParkingSession {
  const ParkingSession({
    required this.id,
    required this.parkingLotId,
    required this.parkingLotName,
    required this.vehiclePlate,
    required this.startedAt,
    required this.paymentStatus,
    this.parkingSpaceCode,
    this.endedAt,
    this.totalAmount,
  });

  final String id;
  final String parkingLotId;
  final String parkingLotName;
  final String? parkingSpaceCode;
  final String vehiclePlate;
  final DateTime? startedAt;
  final DateTime? endedAt;
  final double? totalAmount;
  final String paymentStatus;

  bool get isActive => endedAt == null;

  factory ParkingSession.fromJson(Map<String, dynamic> json) => ParkingSession(
        id: json['id']?.toString() ?? '',
        parkingLotId: json['parkingLotId']?.toString() ?? '',
        parkingLotName: json['parkingLotName']?.toString() ?? '',
        parkingSpaceCode: json['parkingSpaceCode']?.toString(),
        vehiclePlate: json['vehiclePlate']?.toString() ?? '',
        startedAt: DateTime.tryParse(json['startedAt']?.toString() ?? ''),
        endedAt: DateTime.tryParse(json['endedAt']?.toString() ?? ''),
        totalAmount: _nullableDouble(json['totalAmount']),
        paymentStatus: json['paymentStatus']?.toString() ?? 'pending',
      );
}

double? _nullableDouble(dynamic value) {
  if (value == null) return null;
  if (value is num) return value.toDouble();
  return double.tryParse(value.toString());
}
